### PR TITLE
Update EventBusServiceBus.cs

### PR DESCRIPTION
**Pull request by i02coroj**
Do not complete messages until actually processed.

### CODE REVIEW

**Comment by CESARDELATORRE** [comment]
Looks like a good improvement while being a pretty simple update. 👍 
@mvelosop Please, test/confirm this PR and merge when sure, ok?
Thanks for the PR! :)

**Comment by i02coroj** [comment]
Welcome! See you in a few weeks in DotNet2018!
________________________________

File: src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs [hunk 1]
```diff
@@ -148,8 +151,9 @@ private Task ExceptionReceivedHandler(ExceptionReceivedEventArgs exceptionReceiv
             return Task.CompletedTask;
         }
 
-        private async Task ProcessEvent(string eventName, string message)
+        private async Task<bool> ProcessEvent(string eventName, string message)
         {
+            var processed = false;
             if (_subsManager.HasSubscriptionsForEvent(eventName))
             {
                 using (var scope = _autofac.BeginLifetimeScope(AUTOFAC_SCOPE_NAME))
```
Comment (Marusyk) [inline src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: What about using try/catch here?

Comment (Marusyk) [inline src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: What about using try/catch here and do not create a new variable?

File: src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs [hunk 2]
```diff
@@ -129,10 +129,13 @@ private void RegisterSubscriptionClientMessageHandler()
                 {
                     var eventName = $"{message.Label}{INTEGRATION_EVENT_SUFIX}";
                     var messageData = Encoding.UTF8.GetString(message.Body);
-                    await ProcessEvent(eventName, messageData);
+                    var processed = await ProcessEvent(eventName, messageData);
                     
                     // Complete the message so that it is not received again.
-                    await _subscriptionClient.CompleteAsync(message.SystemProperties.LockToken);
+                    if (processed)
```
Comment (Marusyk) [inline src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: Can we use here something like
`if (await ProcessEventAsync(eventName, messageData))`

Comment (i02coroj) [reply src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: Yep, you could.
I just created the variable to improve readability.

File: src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs (hunk 1 above)
Comment (i02coroj) [reply src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: Yep, you could.
I just created the variable to improve readability.

Comment (Marusyk) [reply src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: Yeah, but I talk about try/catch

File: src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs (hunk 2 above)
Comment (Marusyk) [reply src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs]: It's looks ok without variable

**Comment by NeilQ** [comment]
Nice pr.  Curious why this pull request hangs so much time. There's a obvious message persistent bug here. If there is no EventHandler subcribed, the channel will still do ack which cause the message been not persisted.

**Comment by mvelosop** [comment]
Done, thanks @i02coroj, @Marusyk.

**Pull request merged**